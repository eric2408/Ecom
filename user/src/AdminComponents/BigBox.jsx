import React from 'react';
import styled from 'styled-components';
import { Link } from "react-router-dom";
import { useEffect, useState } from "react";
import { userRequest } from "../requestAxios";
import moment from "moment";

const Container = styled.div`
    flex: 2;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
    padding: 20px;
`
const Title = styled.h3`
    font-size: 22px;
    font-weight: 600;
`
const Table = styled.table`
    width: 100%;
    border-spacing: 20px;
`
const TR = styled.tr`
`
const TH = styled.th`
    text-align: left;
`
const User= styled.td`
    display: flex;
    align-items: center;
    font-weight: 600;
`
const Img = styled.img`
    width: 40px;
    height: 40px;
    border-radius: 50%;
    object-fit: cover;
    margin-right: 10px;
`
const Username = styled.span`
`
const Dates = styled.td`
    font-weight: 300;
`
const Amount = styled.td`
    font-weight: 300;
`

const Status = styled.td`
`

const Button = styled.button`
  padding: 8px 9px;
  border: none;
  border-radius: 10px;
  color: black;
  background-color: ${props => props.type === 'success' && '#abd98c'};
  background-color: ${props => props.type === 'pending' && '#9bc4e2'};
  background-color: ${props => props.type === 'cancelled' && '#d07a7a'};
`


function BigBox() {

  const [orders, setOrders] = useState([]);


  useEffect(() => {
    const getOrders = async () => {
      try {
        const res = await userRequest.get("orders/?new=true");
        setOrders(res.data);
      } catch {}
    };
    getOrders();
  }, []);
  

      return (
        <Container>
          <Link to="/admin/orders" style={{textDecoration: 'none', color: 'black'}}>
            <Title>Latest Orders</Title>
          </Link>
          <Table>
            <thead>
            <TR>
              <TH>Customer</TH>
              <TH>Date</TH>
              <TH>Amount</TH>
              <TH>Status</TH>
            </TR>
            </thead>
            <tbody>
            {orders.map((order) => (
            <TR>
              <User key={order._id}>
                <Img src={order.img ||"https://t4.ftcdn.net/jpg/00/64/67/63/360_F_64676383_LdbmhiNM6Ypzb3FM4PPuFP9rHe7ri8Ju.jpg"} />
                <Username>{order.username}</Username>
              </User>
              <Dates>{moment(`${order.createdAt}`).utc().format('YYYY-MM-DD')}</Dates>
              <Amount>${order.amount}</Amount>
              <Status>
                <Button type={order.status}>{order.status}</Button>
              </Status>
            </TR>
            ))}
            </tbody>
          </Table>
        </Container>
      )
}

export default BigBox