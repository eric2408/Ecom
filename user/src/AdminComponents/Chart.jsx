import React from 'react'
import styled from 'styled-components';
import { LineChart, Line, XAxis, CartesianGrid, Tooltip, ResponsiveContainer } from "recharts";

const Container = styled.div`
    margin: 30px;
    padding: 20px;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
`

const Title = styled.h3`
    margin: 20px;
`

function Chart({ title, data, dataKey, grid }) {
  return (
    <Container>
      <Title>{title}</Title>
      <ResponsiveContainer width="100%" aspect={4 / 1}>
        <LineChart data={data}>
          <XAxis dataKey="name" stroke="black" />
          <Line type="monotone" dataKey={dataKey} stroke="#6f50e6" />
          <Tooltip />
          {grid && <CartesianGrid stroke="#e0dfdf" />}
        </LineChart>
      </ResponsiveContainer>
    </Container>
  )
}

export default Chart