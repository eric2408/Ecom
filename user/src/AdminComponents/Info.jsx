import React from 'react';
import styled from 'styled-components';
import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';

const Container = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
`

const Wrapper = styled.div`
    flex: 1;
    margin: 0px 20px;
    padding: 30px;
    border-radius: 10px;
    cursor: pointer;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
`

const Title = styled.span`
    font-size: 20px;
`

const InfoContainer = styled.div`
    margin: 10px 0px;
    display: flex;
    align-items: center;
`

const Amount = styled.span`
    font-size: 30px;
    font-weight: 600;
`

const Rate = styled.span`
    display: flex;
    align-items: center;
    margin-left: 20px;
`


const NegativeIcon = styled.div`
    font-size: 14px;
    margin-left: 5px;
    color: red;
`

const PositiveIcon = styled.div`
    font-size: 14px;
    margin-left: 5px;
    color: green;
`

const Desc = styled.span`
    font-size: 15px;
    color: gray;
`



function Info() {
  return (
    <Container>
    <Wrapper>
      <Title>Revenue</Title>
      <InfoContainer>
        <Amount>$4,415</Amount>
        <Rate>
          -11.4 
          <NegativeIcon>
            <ArrowDownwardIcon/>
          </NegativeIcon>
        </Rate>
      </InfoContainer>
      <Desc>Compared to last month</Desc>
    </Wrapper>
    <Wrapper>
      <Title>Profit</Title>
      <InfoContainer>
        <Amount>$2,415</Amount>
        <Rate>
          -1.4 
          <NegativeIcon>
            <ArrowDownwardIcon/>
          </NegativeIcon>
        </Rate>
      </InfoContainer>
      <Desc>Compared to last month</Desc>
    </Wrapper>
    <Wrapper>
      <Title>Cost</Title>
      <InfoContainer>
        <Amount>$2,000</Amount>
        <Rate>
          +2.4 
          <PositiveIcon>
            <ArrowUpwardIcon/>
          </PositiveIcon>
        </Rate>
      </InfoContainer>
      <Desc>Compared to last month</Desc>
    </Wrapper>
  </Container>
  )
}

export default Info