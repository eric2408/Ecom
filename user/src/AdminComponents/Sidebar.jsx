import React from 'react';
import styled from 'styled-components';
import { Link } from "react-router-dom";
import ArticleIcon from '@mui/icons-material/Article';
import TimelineIcon from '@mui/icons-material/Timeline';
import PaidIcon from '@mui/icons-material/Paid';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import Inventory2Icon from '@mui/icons-material/Inventory2';
import ReceiptLongIcon from '@mui/icons-material/ReceiptLong';
import AssessmentIcon from '@mui/icons-material/Assessment';
import MailOutlineIcon from '@mui/icons-material/MailOutline';
import MessageIcon from '@mui/icons-material/Message';

const Container = styled.div`
    flex: 1;
    height: calc(100vh - 50px);
    background-color:#E7E3E8;
    position: sticky;
    top: 50px;
`

const Wrapper = styled.div`
    padding: 20px;
    color: black;
`

const BarContainer = styled.div`
    margin-bottom: 10px;
    margin-left: 25px;
    margin-top: 45px;
`

const Title = styled.h3`
    font-size: 20px;
    color: black;
    margin-bottom: 10px;
`

const List = styled.ul`
    list-style: none;
    padding: 5px;
`

const ActiveItem = styled.li`
    padding: 5px;
    cursor: pointer;
    display: flex;
    align-items: center;
    border-radius: 10px;
    background-color: #E7E3E8;
    :hover {
        color:white;
        background-color: #6f50e6;
    }
`

const Item = styled.li`
    padding: 5px;
    cursor: pointer;
    display: flex;
    align-items: center;
    border-radius: 10px;
    :hover {
        color:white;
        background-color: #6f50e6;
    }
`

const Icon = styled.div`
    margin-right: 5px;
    font-size: 20px !important;
`

function Sidebar() {
  return (
    <Container>
    <Wrapper>
      <BarContainer>
        <Title>Dashboard</Title>
        <List>
          <Link to="/admin" style={{textDecoration: 'none', color: 'black'}}>
          <ActiveItem>
            <Icon>
            <ArticleIcon />
            </Icon>
            Home
          </ActiveItem>
          </Link>
          <Item>
            <Icon>
                <TimelineIcon />
            </Icon>
            Analytics
          </Item>
          <Item>
            <Icon>
                <PaidIcon />
            </Icon>
            Net Profit
          </Item>
        </List>
      </BarContainer>
      <BarContainer>
        <Title>Manage</Title>
        <List>
          <Link to="/admin/products/" style={{textDecoration: 'none', color: 'black'}}>
            <Item>
                <Icon>
                    <Inventory2Icon />
                </Icon>
              Products
            </Item>
          </Link>
          <Link to="/admin/orders" style={{textDecoration: 'none', color: 'black'}}>
            <Item>
              <Icon>
                  <ReceiptLongIcon />
              </Icon>
              Orders
            </Item>
          </Link> 
            <Item>
            <Icon>
                <AssessmentIcon />
            </Icon>
            Reports
          </Item>
        </List>
      </BarContainer>
      <BarContainer>
        <Title>Notifications</Title>
        <List>
          <Item>
            <Icon>
                <MailOutlineIcon />
            </Icon>
            Mail
          </Item>
          <Item>
            <Icon>
                <MessageIcon />
            </Icon>
            Messages
          </Item>
        </List>
      </BarContainer>
    </Wrapper>
  </Container>
  )
}

export default Sidebar