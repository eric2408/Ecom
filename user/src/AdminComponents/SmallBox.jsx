import React from 'react'
import styled from 'styled-components';

const Container = styled.div`
    flex: 1;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
    padding: 20px;
    margin-right: 20px;
`

const Title = styled.span`
    font-size: 22px;
    font-weight: 600;
`
const UL = styled.ul`
    margin: 0;
    padding: 0;
    list-style: none;
`

const List = styled.li`
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 20px 0px;
`

const Img = styled.img`
    width: 40px;
    height: 40px;
    border-radius: 50%;
    object-fit: cover;
`

const User = styled.div`
    display: flex;
    flex-direction: column;
`

const Username = styled.span`
    font-weight: 600;
`

const UserJob = styled.span`
    font-weight: 300;
`

const Button = styled.div`
    display: flex;
    align-items: center;
    border: none;
    border-radius: 10px;
    padding: 7px 10px;
    background-color: #6f50e6;
    color: white;
    cursor: pointer;
`



function SmallBox() {
  return (
    <Container>
      <Title>New Join Members</Title>
      <UL>
        <List>
          <Img src="https://images.pexels.com/photos/3992656/pexels-photo-3992656.png?auto=compress&cs=tinysrgb&dpr=2&w=500"/>
          <User>
            <Username>Joe Rogan</Username>
            <UserJob>Software Engineer</UserJob>
          </User>
          <Button>
            Display
          </Button>
        </List>
        <List>
            <Img src="https://images.pexels.com/photos/3992656/pexels-photo-3992656.png?auto=compress&cs=tinysrgb&dpr=2&w=500"/>
          <User>
            <Username>Anna Keller</Username>
            <UserJob>Software Engineer</UserJob>
          </User>
          <Button>
            Display
          </Button>
        </List>
        <List>
            <Img src="https://images.pexels.com/photos/3992656/pexels-photo-3992656.png?auto=compress&cs=tinysrgb&dpr=2&w=500"/>
          <User>
            <Username>Anna Keller</Username>
            <UserJob>Software Engineer</UserJob>
          </User>
          <Button>
            Display
          </Button>
        </List>
        <List>
            <Img src="https://images.pexels.com/photos/3992656/pexels-photo-3992656.png?auto=compress&cs=tinysrgb&dpr=2&w=500"/>
          <User>
            <Username>Anna Keller</Username>
            <UserJob>Software Engineer</UserJob>
          </User>
          <Button>
            Display
          </Button>
        </List>
        <List>
            <Img src="https://images.pexels.com/photos/3992656/pexels-photo-3992656.png?auto=compress&cs=tinysrgb&dpr=2&w=500"/>
          <User>
            <Username>Anna Keller</Username>
            <UserJob>Software Engineer</UserJob>
          </User>
          <Button>
            Display
          </Button>
        </List>
      </UL>
    </Container>
  )
}

export default SmallBox