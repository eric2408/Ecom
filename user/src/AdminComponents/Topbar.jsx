import React from 'react'
import NotificationsNoneIcon from '@mui/icons-material/NotificationsNone';
import { Badge } from '@mui/material';
import GTranslateIcon from '@mui/icons-material/GTranslate';
import SettingsIcon from '@mui/icons-material/Settings';
import styled from 'styled-components';
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";

const Container = styled.div`
    width: 100%;
    height: 100px;
    background-color: #14141f;
    position: sticky;
    top: 0;
    z-index: 999;
`

const TopbarWrapper = styled.div`
    height: 100%;
    padding: 0px 40px;
    display: flex;
    align-items: center;
    justify-content: space-between;
`
const Logo = styled.h1`
  font-weight: bold;
  color: #6f50e6;
`;

const TopRight = styled.div`
    display: flex;
    align-items: center;
`
const IconContainer = styled.div`
    position: relative;
    cursor: pointer;
    margin-right: 40px;
    color: white;
    :hover{
      color: #6f50e6;
    }
`


const Img = styled.img`
    width: 50px;
    height: 50px;
    border-radius: 50%;
    cursor: pointer;
`

function Topbar() {
  const user = useSelector((state) => state.user.currentUser);

  return (
    <Container>
    <TopbarWrapper>
      <div>
        <Link to="/" style={{textDecoration: 'none'}}>
          <Logo>Virtual Space</Logo>
        </Link>
      </div>
      <TopRight>
        <IconContainer>
          <GTranslateIcon />
        </IconContainer>
        <IconContainer>
            <Badge badgeContent={2} color="secondary">
                <NotificationsNoneIcon/>
            </Badge>
        </IconContainer>
        <IconContainer>
            <Badge badgeContent={2} color="secondary">
                <SettingsIcon />
            </Badge>
        </IconContainer>
        <Img src={user.img}/>
      </TopRight>
    </TopbarWrapper>
  </Container>
  )
}

export default Topbar