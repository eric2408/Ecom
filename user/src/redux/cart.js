import {createSlice} from '@reduxjs/toolkit';


const cart = createSlice({
    name:"cart",
    initialState:{
        products: [],
        quantity:0,
        total:0
    },
    reducers:{
        addProduct: (state, action) => {
            state.quantity += action.payload.quantity;
            state.products.push(action.payload)
            state.total += action.payload.price * action.payload.quantity;
        },
        deleteProduct: (state, action) => {
            state.products= [];
            state.quantity= 0;
            state.total= 0;
        },
        addOne: (state, action) => {
            state.quantity += 1;
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0) {
                state.products[index].quantity += 1;
            }
        },
        subtractOne: (state, action) => {
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0 && state.products[index].quantity > 0) {
                state.products[index].quantity -= 1;
                state.quantity -= 1;
            }
        },
        removeOne:(state, action) => {
            const index = state.products.findIndex(
                    (item) => item._id === action.payload.id
                );
            if (index >= 0) {
                    state.quantity -= state.products[index].quantity;

                    state.products.splice(index, 1);
            } else {
                    console.warn(
                      `Cant remove product (id: ${action.payload.id}) as its not in basket!`
                    )
            }
        }
    }
})

export const {addProduct,  deleteProduct, addOne, subtractOne, removeOne} = cart.actions

export default cart.reducer;

export const emptyCart = async(dispatch) => {
    try{
      dispatch(deleteProduct());
    } catch(e){
      console.log(e)
    }
  }

