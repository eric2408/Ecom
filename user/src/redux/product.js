import { createSlice } from "@reduxjs/toolkit";
import { userRequest } from "../requestAxios";

export const productSlice = createSlice({
  name: "product",
  initialState: {
    products: [],
    isFetching: false,
    error: false,
  },
  reducers: {
    //Get all products
    getProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    getProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products = action.payload;
    },
    getProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Delete Products
    deleteProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    deleteProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.splice(state.products.findIndex((item) => item._id === action.payload), 1);
    },
    deleteProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Update products
    updateProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    updateProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products[
        state.products.findIndex((item) => item._id === action.payload.id)
      ] = action.payload.product;
    },
    updateProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Add Product
    addProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    addProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.push(action.payload);
    },
    addProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
  },
});

export const {
  getProductStart,
  getProductSuccess,
  getProductFailure,
  deleteProductStart,
  deleteProductSuccess,
  deleteProductFailure,
  updateProductStart,
  updateProductSuccess,
  updateProductFailure,
  addProductStart,
  addProductSuccess,
  addProductFailure,
} = productSlice.actions;

export default productSlice.reducer;

export const getProducts = async (dispatch) => {
    dispatch(getProductStart());
    try {
      const res = await userRequest.get("/products");
      dispatch(getProductSuccess(res.data));
    } catch (err) {
      dispatch(getProductFailure());
    }
  };


export const deleteProduct = async (id, dispatch) => {
  dispatch(deleteProductStart());
  try {
    const res = await userRequest.delete(`/products/${id}`);
    dispatch(deleteProductSuccess(id));
  } catch (err) {
    dispatch(deleteProductFailure());
  }
};

export const updateProduct = async (id, product, dispatch) => {
  dispatch(updateProductStart());
  try {
    const res = await userRequest.put(`/products/${id}`, product);
    dispatch(updateProductSuccess({ id, product }));
  } catch (err) {
    dispatch(updateProductFailure());
  }
};
export const addProduct = async (product, dispatch) => {
  dispatch(addProductStart());
  try {
    const res = await userRequest.post(`/products`, product);
    dispatch(addProductSuccess(res.data));
  } catch (err) {
    dispatch(addProductFailure());
  }
};