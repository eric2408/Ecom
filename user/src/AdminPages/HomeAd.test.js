import React from "react";
import { render } from "@testing-library/react";
import Topbar from '../AdminComponents/Topbar';
import Sidebar from '../AdminComponents/Sidebar';
import Info from '../AdminComponents/Info';
import Chart from '../AdminComponents/Chart';
import SmallBox from '../AdminComponents/SmallBox';
import BigBox from '../AdminComponents/BigBox';
import { BrowserRouter } from "react-router-dom";
import {store, persistor} from '../redux/store';
import{Provider} from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
            <Sidebar />
            <Info/> 
            <SmallBox />
            <BigBox />
      </BrowserRouter>
    </PersistGate>
  </Provider>

  );
  expect(asFragment()).toMatchSnapshot();
});