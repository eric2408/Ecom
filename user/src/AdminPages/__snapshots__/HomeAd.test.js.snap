// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gKPRtg BdipU"
  >
    <div
      class="sc-iBYQkv AOdRb"
    >
      <div
        class="sc-ftTHYK bPjZpQ"
      >
        <h3
          class="sc-pyfCe lnaBVY"
        >
          Dashboard
        </h3>
        <ul
          class="sc-jrcTuL eEEtTg"
        >
          <a
            href="/admin"
            style="text-decoration: none; color: black;"
          >
            <li
              class="sc-kDvujY hQvcpZ"
            >
              <div
                class="sc-csuSiG jbkwuc"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArticleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                  />
                </svg>
              </div>
              Home
            </li>
          </a>
          <li
            class="sc-ipEyDJ cxvbdy"
          >
            <div
              class="sc-csuSiG jbkwuc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="TimelineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
                />
              </svg>
            </div>
            Analytics
          </li>
          <li
            class="sc-ipEyDJ cxvbdy"
          >
            <div
              class="sc-csuSiG jbkwuc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PaidIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm.88 15.76V19h-1.75v-1.29c-.74-.18-2.39-.77-3.02-2.96l1.65-.67c.06.22.58 2.09 2.4 2.09.93 0 1.98-.48 1.98-1.61 0-.96-.7-1.46-2.28-2.03-1.1-.39-3.35-1.03-3.35-3.31 0-.1.01-2.4 2.62-2.96V5h1.75v1.24c1.84.32 2.51 1.79 2.66 2.23l-1.58.67c-.11-.35-.59-1.34-1.9-1.34-.7 0-1.81.37-1.81 1.39 0 .95.86 1.31 2.64 1.9 2.4.83 3.01 2.05 3.01 3.45 0 2.63-2.5 3.13-3.02 3.22z"
                />
              </svg>
            </div>
            Net Profit
          </li>
        </ul>
      </div>
      <div
        class="sc-ftTHYK bPjZpQ"
      >
        <h3
          class="sc-pyfCe lnaBVY"
        >
          Manage
        </h3>
        <ul
          class="sc-jrcTuL eEEtTg"
        >
          <a
            href="/admin/products/"
            style="text-decoration: none; color: black;"
          >
            <li
              class="sc-ipEyDJ cxvbdy"
            >
              <div
                class="sc-csuSiG jbkwuc"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="Inventory2Icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4h16v3z"
                  />
                </svg>
              </div>
              Products
            </li>
          </a>
          <a
            href="/admin/orders"
            style="text-decoration: none; color: black;"
          >
            <li
              class="sc-ipEyDJ cxvbdy"
            >
              <div
                class="sc-csuSiG jbkwuc"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ReceiptLongIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.5 3.5 18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2v14H3v3c0 1.66 1.34 3 3 3h12c1.66 0 3-1.34 3-3V2l-1.5 1.5zM19 19c0 .55-.45 1-1 1s-1-.45-1-1v-3H8V5h11v14z"
                  />
                  <path
                    d="M9 7h6v2H9zm7 0h2v2h-2zm-7 3h6v2H9zm7 0h2v2h-2z"
                  />
                </svg>
              </div>
              Orders
            </li>
          </a>
          <li
            class="sc-ipEyDJ cxvbdy"
          >
            <div
              class="sc-csuSiG jbkwuc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AssessmentIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
                />
              </svg>
            </div>
            Reports
          </li>
        </ul>
      </div>
      <div
        class="sc-ftTHYK bPjZpQ"
      >
        <h3
          class="sc-pyfCe lnaBVY"
        >
          Notifications
        </h3>
        <ul
          class="sc-jrcTuL eEEtTg"
        >
          <li
            class="sc-ipEyDJ cxvbdy"
          >
            <div
              class="sc-csuSiG jbkwuc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MailOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                />
              </svg>
            </div>
            Mail
          </li>
          <li
            class="sc-ipEyDJ cxvbdy"
          >
            <div
              class="sc-csuSiG jbkwuc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MessageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
                />
              </svg>
            </div>
            Messages
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="sc-eDWCr fQEuTe"
  >
    <div
      class="sc-bqWxrE iVirQj"
    >
      <span
        class="sc-ksBlkl ksIERD"
      >
        Revenue
      </span>
      <div
        class="sc-hBxehG dlChPQ"
      >
        <span
          class="sc-fnGiBr bYdify"
        >
          $
        </span>
        <span
          class="sc-fEXmlR gA-DJKZ"
        >
          0% 
          <div
            class="sc-idXgbr ieXuuo"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowUpwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
          </div>
        </span>
      </div>
      <span
        class="sc-dIfARi gHXcEm"
      >
        Compared to last month
      </span>
    </div>
    <div
      class="sc-bqWxrE iVirQj"
    >
      <span
        class="sc-ksBlkl ksIERD"
      >
        Profit
      </span>
      <div
        class="sc-hBxehG dlChPQ"
      >
        <span
          class="sc-fnGiBr bYdify"
        >
          $NaN
        </span>
        <span
          class="sc-fEXmlR gA-DJKZ"
        >
          -14%
          <div
            class="sc-bjfHbI kRJUbL"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDownwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </div>
        </span>
      </div>
      <span
        class="sc-dIfARi gHXcEm"
      >
        Compared to last month
      </span>
    </div>
    <div
      class="sc-bqWxrE iVirQj"
    >
      <span
        class="sc-ksBlkl ksIERD"
      >
        Cost
      </span>
      <div
        class="sc-hBxehG dlChPQ"
      >
        <span
          class="sc-fnGiBr bYdify"
        >
          $1500
        </span>
        <span
          class="sc-fEXmlR gA-DJKZ"
        >
          +32%
          <div
            class="sc-idXgbr ieXuuo"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowUpwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
          </div>
        </span>
      </div>
      <span
        class="sc-dIfARi gHXcEm"
      >
        Compared to last month
      </span>
    </div>
  </div>
  <div
    class="sc-kgTSHT gUKEoW"
  >
    <span
      class="sc-fLcnxK huhDxH"
    >
      Your Customers
    </span>
    <ul
      class="sc-bBABsx bneAqS"
    />
  </div>
  <div
    class="sc-ezOQGI bfZQHT"
  >
    <a
      href="/admin/orders"
      style="text-decoration: none; color: black;"
    >
      <h3
        class="sc-bYMpWt cjLmim"
      >
        Latest Orders
      </h3>
    </a>
    <table
      class="sc-kMjNwy cQjMKw"
    >
      <thead>
        <tr
          class="sc-ilhmMj jgQzif"
        >
          <th
            class="sc-jIRcFI dwa-DLU"
          >
            Customer
          </th>
          <th
            class="sc-jIRcFI dwa-DLU"
          >
            Date
          </th>
          <th
            class="sc-jIRcFI dwa-DLU"
          >
            Amount
          </th>
          <th
            class="sc-jIRcFI dwa-DLU"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
